articles/200_migration_guide_from_ros1.md

```diff
@@ -162,3 +162,483 @@ NOTE: to be written
+  <exec_depend>rclcpp</exec_depend>
+  <exec_depend>rmw_implementation</exec_depend>
+<!--  <run_depend>std_msgs</run_depend> -->
+  <exec_depend>std_msgs</exec_depend>
+~~~
+
+We also need to tell `ament` what *kind* of package we are, so that it knows how
+to build us.  Because we're using `ament` and CMake, we add the following lines
+to declare our build type to be `ament_cmake`:
+
+~~~
+  <export>
+    <build_type>ament_cmake</build_type>
+  </export>
+~~~
+
+
+Putting it all together, our `package.xml` now looks like this:
+
+~~~
+<!-- <package> -->
+<package format="2">
+  <name>talker</name>
+  <version>0.0.0</version>
+  <description>talker</description>
+  <maintainer email="gerkey@osrfoundation.org">Brian Gerkey</maintainer>
+  <license>Apache License 2.0</license>
+<!--  <buildtool_depend>catkin</buildtool_depend> -->
+  <buildtool_depend>ament_cmake</buildtool_depend>
+<!--  <build_depend>roscpp</build_depend> -->
+  <build_depend>rclcpp</build_depend>
+  <build_depend>rmw_implementation</build_depend>
+  <build_depend>std_msgs</build_depend>
+<!--  <run_depend>roscpp</run_depend> -->
+  <exec_depend>rclcpp</exec_depend>
+  <exec_depend>rmw_implementation</exec_depend>
+<!--  <run_depend>std_msgs</run_depend> -->
+  <exec_depend>std_msgs</exec_depend>
+  <export>
+    <build_type>ament_cmake</build_type>
+  </export>
+</package>
+~~~
+
+*TODO: show simpler version of this file just using the `<depend>` tag, which is
+enabled by version 2 of the package format (also supported in `catkin` so,
+strictly speaking, orthogonal to ROS 2).*
+
+#### Changing the CMake code
+
+ROS 2 relies on the C++11 standard.  Depending on what compiler you're using,
+support for C++11 might not be enabled by default.  Using `gcc` 4.8 (which is
+what is used on Ubuntu Trusty), we need to enable it explicitly, which we do
+by adding this line near the top of the file:
+
+~~~
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+~~~
+
+Using `catkin`, we specify the packages we want to build against by passing them
+as `COMPONENTS` arguments when initially finding `catkin` itself.  With `ament`,
+we find each package individually, starting with `ament_cmake` (and adding our
+new dependency, `rmw_implementation`):
+
+~~~
+#find_package(catkin REQUIRED COMPONENTS roscpp std_msgs)
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rmw_implementation REQUIRED)
+find_package(std_msgs REQUIRED)
+~~~
+
+We call `catkin_package()` to auto-generate things like CMake configuration
+files for other packages that use our package.  Whereas that call happens
+*before* specifying targets to build, we now call the analogous
+`ament_package()` *after* the targets:
+
+~~~
+#catkin_package()
+# At the bottom of the file:
+ament_package()
+~~~
+
+Similarly to how we found each dependent package separately, instead of finding
+them as parts of catkin, we also need to add their include directories
+separately:
```

Comment:
Maybe mention that for this example it doesn't make a different but for more complex example and ament_auto it does?